### DISCUSSION TITLE
Is it possible to execute a command in the directory `just` has been invoked in?

### DISCUSSION THREAD

**Discussion by koehlma** [casey/just#1360] [Q&A] [answered]
By default, commands are executed in the directory the `justfile` is located in (which makes sense most of the time). Is it possible to prevent this behavior and have a command execute in the directory `just` has been invoked in instead?

**Comment by casey** [comment, marked as answer]
Good question!

You could do this:

```
foo:
  cd {{invocation_directory()}} && bar
```

So `just foo` would run `bar` in the invocation directory

**Reply by koehlma** [reply]
Amazing! Thank you so much for answering the question and developing `just`. 👍

**Reply by casey** [reply]
Glad that helps! And you are most welcome!

**Reply by davised** [reply]
Hi Casey,

How difficult would it be to implement a flag to start all recipes in the invocation directory? Sometimes I forget that justfiles don't work like bash scripts that start in the working directory.

I'll typically do this when necessary:

```make
#!/usr/bin/env just
wd := invocation_directory()

foo:
  #!/usr/bin/env bash
  cd {{wd}}
  ...
```
which isn't too bad, but I will say that sometimes it would be nice to have that option. And certainly some folks who are used to bash scripting will more easily convert to justfiles for convenience if there's a way to force it to work the same. 

I'll typically just make a symlink in the working directory so I don't have to deal with the perceived limitation of the software.

At any rate, I understand if there isn't a desire or if it's too difficult, but I figured I'd put it out there.

**Reply by casey** [reply]
I wouldn't be opposed to something like that. Definitely comment on #762!

**Comment by runeimp** [comment]
I like `--invokedir` as a command line flag for such a facility.  👼

**Reply by mahiki** [reply]
This is really key, otherwise I'm putting `[no-cd]` over every single recipe. its that or symlink the justfile into every subproject folder.

I'd really like to have a `justfile` level attribute like `set no-cd := true`
